PowerPointLabs/PowerPointLabs/SpeechEngine/TextToSpeech.cs

```diff
@@ -59,11 +68,106 @@ public static void SpeakString(String textToSpeak)
             PromptToAudio.Speak(builder);
         }
 
+        private static void SaveStringToWaveFileWithHumanVoice(string textToSave, string filePath)
+        {
+            string accessToken;
+            string textToSpeak = GetHumanSpeakNotesForText(textToSave);
+
+            try
+            {
+                Authentication auth = Authentication.GetInstance();
+                accessToken = auth.GetAccessToken();
+                Logger.Log("Token: " + accessToken);

```

Comment:
Account authentication is performed before sending http request. If user account has expired or invalid, an error message will be displayed to user.